### PR TITLE
firefox: add handlers.json configuration

### PR DESCRIPTION
**Pull request by kugland**
### Description

Adds support for configuring Firefox's handlers.json file to manage MIME type and URL scheme handlers declaratively (at `programs.firefox.profiles.<profile>.handlers`). Handlers control how Firefox opens files and protocols (e.g., PDF viewers, mailto handlers).

The implementation validates handler configurations and generates properly formatted JSON output. Includes comprehensive test coverage for edge cases and assertions.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -p treefmt nixfmt deadnix keep-sorted --run treefmt`.

- [x] Code tested through `nix run .#tests -- test-all` or
      `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [x] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [x] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [x] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

### CODE REVIEW

**Comment by kugland** [comment]
Closes https://github.com/nix-community/home-manager/issues/5617

File: tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+              handlers = [
+                {
+                  name = "Viewer";
+                  path = "${pkgs.hello}/bin/hello";
+                }
+                {
+                  name = "Editor";
+                  path = "${pkgs.hello}/bin/hello";
+                }
+              ];
```
Comment (teto) [inline tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix:34]: what happens in firefox when there are several handlers ? Does it ask the user to choose ?

**Comment by teto** [comment]
could be a nice addition. How is this different than setting the policies as shown in this comment https://github.com/nix-community/home-manager/issues/5617#issuecomment-2294170015 ?

**Comment by kugland** [comment]
> could be a nice addition. How is this different than setting the policies as shown in this comment [#5617 (comment)](https://github.com/nix-community/home-manager/issues/5617#issuecomment-2294170015) ?

This is per profile, and it does sanity checking with the config; policies are global and are not checked.

Furthermore, Home Manager currently has `programs.firefox.profiles.<name>.bookmarks`, `programs.firefox.profiles.<name>.containers` and `programs.firefox.profiles.<name>.search`, all of which can be set through policies also.

File: tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix (hunk 1 above)
Comment (kugland) [reply tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix:34]: If ask = true, yes. If it’s not, it goes with the first handler.

File: modules/programs/firefox/profiles/handlers.nix [hunk 2]
```diff
@@ -0,0 +1,283 @@
+{
+  config,
+  lib,
+  pkgs,
+  appName,
+  package,
+  modulePath,
+  profilePath,
+}:
+with lib;
```
Comment (khaneliman) [inline modules/programs/firefox/profiles/handlers.nix]:
```suggestion
```

File: tests/modules/programs/firefox/profiles/handlers/assertions.nix [hunk 3]
```diff
@@ -0,0 +1,134 @@
+modulePath:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
```
Comment (khaneliman) [inline tests/modules/programs/firefox/profiles/handlers/assertions.nix]:
```suggestion
```

File: tests/modules/programs/firefox/profiles/handlers/default.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
+modulePath:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
```
Comment (khaneliman) [inline tests/modules/programs/firefox/profiles/handlers/default.nix]:
```suggestion
```

File: tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
+modulePath:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
```
Comment (khaneliman) [inline tests/modules/programs/firefox/profiles/handlers/multiple-handlers.nix]:
```suggestion
```

File: tests/modules/programs/firefox/profiles/handlers/no-default-handler.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
+modulePath:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
```
Comment (khaneliman) [inline tests/modules/programs/firefox/profiles/handlers/no-default-handler.nix]:
```suggestion
```

File: tests/modules/programs/firefox/profiles/handlers/path-or-uri-template.nix [hunk 7]
```diff
@@ -0,0 +1,57 @@
+modulePath:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
```
Comment (khaneliman) [inline tests/modules/programs/firefox/profiles/handlers/path-or-uri-template.nix]:
```suggestion
```

**Comment by khaneliman** [comment]
Please recreate.
